src/components/QuizGame/index.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import { fetchQuizQuestions } from "../../components/QuizGameApi";
+import "antd/dist/antd.css";
+import QuestionCard from "../../components/QuizGameQuestions";
+import "./index.css";
+const Difficulty = {
+  EASY: "easy",
+  MEDIUM: "medium",
+  HARD: "hard",
+};
+const TOTAL_QUESTIONS = 10;
+const QuizGame = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionNumber, setCurrentQuestionNumber] = useState(0);
+  const [userAnswers, setUserAnswers] = useState([]);
+  const [score, setScore] = useState(0);
+  const [isGameOver, setIsGameOver] = useState(true);
+  const startQuiz = async () => {
+    setIsLoading(true);
+    setIsGameOver(false);
+    const newQuestions = await fetchQuizQuestions(
+      TOTAL_QUESTIONS,
+      Difficulty.EASY
+    );
+    setQuestions(newQuestions);
+    setScore(0);
+    setUserAnswers([]);
+    setCurrentQuestionNumber(0);
+    setIsLoading(false);
+  };
+  const checkAnswer = (e) => {
+    if (!isGameOver) {
+      const answer = e.currentTarget.value;
+      const isCorrect = questions[currentQuestionNumber].correct_answer === answer;
+      if (isCorrect) {
+        setScore((prev) => prev + 1);
+        const answerObject = {
+          question: questions[currentQuestionNumber].question,
+          answer,
+          isCorrect,
+          correctAnswer: questions[currentQuestionNumber].correct_answer,
+        };
+        setUserAnswers((prev) => [...prev, answerObject]);
+      }
+    }
+  };
+  const nextQuestion = () => {
+    const nextQ = currentQuestionNumber + 1;
+    if (nextQ >= TOTAL_QUESTIONS) {
+      setIsGameOver(true);
+    } else {
+      setCurrentQuestionNumber(nextQ);
+    }
+  };
+  const checkGameOver = isGameOver || userAnswers.length === TOTAL_QUESTIONS;
+  const  checkTonextQuestion = !isGameOver &&!isLoading && userAnswers.length === currentQuestionNumber+ 1 && currentQuestionNumber !== TOTAL_QUESTIONS - 1 
```

Comment:
the idea of this that the (next question) button will not appear until you answer the question, it means you cant skip a question without answering, that why it has conditions, the conditions are you clicked on start button, you answer the question, you didnt finish the 10 question, otherwise you can see this button . and for deceleration it already declared , check to next question